content/blog/2020-03-tibble-3-0-0/index.Rmarkdown

```diff
@@ -0,0 +1,185 @@
+Due to the nature of the changes, about 60 CRAN packages were failing with our release candidate.
+Thanks to the maintainers of downstream packages who were very helpful in making this upgrade a smooth experience.
+
+In this blog post, I focus on a few user- and programmer-related changes, and give an outlook over future development:
+
+- How to ensure that an object can be part of a tibble
+- Sturdy recycling
+- Pitfalls with type conversion
+- Safe error handling
+- Formatting
+
+For a complete overview please see the [release notes](https://github.com/tidyverse/tibble/releases/tag/v2.0.0).
+
+Use the [issue tracker](https://github.com/tidyverse/tibble/issues) to submit bugs or suggest ideas, your contributions are always welcome.
+
+
+## What can be part of a tibble?
+
+Tibbles and data frames are collections of columns, each is a vector, all have the same length.
+Neat.
+
+What is a vector?
+What is its length?
+
+The new [vctrs package](https://vctrs.r-lib.org) is dedicated to answering these, surprisingly tricky, questions.
+The `vctrs::vec_is()` function decides if an object is a vector.
+This is important, because some objects are inherently scalar and cannot be added as a column to a data frame.
+
+Environments, functions, and other "special" types of objects are clearly non-vectors.
+It is far less obvious for lists, in particular if they have a `"class"` attribute:
+
+```{r compare-list}
+model <- lm(y ~ x, data.frame(x = 1:3, y = 2:4), model = FALSE)
+is.list(model)
+length(model)
+time <- as.POSIXlt(Sys.time())
+is.list(time)
+length(time)
+length(unclass(time))
+```
+
+By relying on `vctrs::vec_is()`, the `tibble::tibble()` function and others can give an early error if used with an inherent scalar:
+
+```{r tibble}
+library(tibble)
+tibble(model)
+tibble(time)
+
+x <- tibble(time)
+x$model <- model
+```
+
+This is safer than the data frame implementation, which currently allows putting all lists (vector or not) into a column:
+
+```{r update-model}
+y <- data.frame(x = 1)
+y$model <- model
+y <- data.frame(x = 1:11)
+y$model <- model
+as_tibble(y)
+```
+
+The new `tibble_row()` function reverses this: inherent scalars are wrapped in lists:
+
+```{r tibble-row}
+tibble_row(model)
+tibble_row(time)
+tibble_row(time = rep(time, 2))
+```
+
+For the `"POSIXlt"` class, this is handled by vctrs internally.
+For your own class, there are two ways to ensure that a list object is treated as a vector:
+
+- Add an explicit `"list"` to its classes
+- Implement a `vctrs::vec_proxy()` method
+
+```{r list}
+x <- structure(list(1), class = c("foo", "list"))
+vctrs::vec_is(x)
+
+y <- structure(list(1), class = "bar")
+vctrs::vec_is(y)
+
+vec_proxy.bar <- function(x, ...) x
+vctrs::vec_is(y)
+```
+
+If you have implemented a vector class (list or not), please add it to my [Awesome vectors](https://github.com/krlmlr/awesome-vctrs#readme) list, or file an issue.
+
+## Sturdy recycling
```

Comment:
I think you're missing a section that talks about size. That would be a useful place to emphasise that data frames can be columns of a data frame.